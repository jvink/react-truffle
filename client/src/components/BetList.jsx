import React, { Component } from "react";
import "../App.css";


class App extends React.Component {
    state = {stackId:null, bets: null};

    componentWillReceiveProps = async (props) => {
      this.setState({bets: props.bets, drizzle: props.drizzle})
    }
  
    setValue = async () => {
        const { drizzle, drizzleState} = this.props;
        const contract = drizzle.contracts.SimpleStorage;
    // let drizzle know we want to call the `add` method with `value1 and value2`
        const stackId = await contract.methods["set"].cacheSend(100 ,{
          from: drizzleState.accounts[0], gas: 2000000
        });
        console.log(stackId)
    // save the `stackId` for later reference
        this.setState({ stackId });
      };
    
    render() {
      if (this.state.loading ) {
        return <div>Loading Drizzle, Web3, Metamask...</div>;
      }

      const {drizzleState, storageValue} = this.props;
      const {bets} = this.state;
  // if it exists, then we display its value
      return (    
            <div className="card col-4">
              <h2>Wallet</h2>
              <p>The stored value is: {storageValue && storageValue}</p>
              <h3>Accounts: </h3>
              <ul>
                <li>{drizzleState && drizzleState.accounts[0] }</li>
              </ul>
              <button type="button" onClick={this.setValue} className="btn btn-secondary mb-4">click here</button>
              <label>Lijst bets (deze nog fatsoeneren)</label>
              {bets && bets.map((bet, key) => {console.log(bet); return <div key={key}>{bet.cityId}</div>})}
            </div>
      );
    }
  }
  export default App;