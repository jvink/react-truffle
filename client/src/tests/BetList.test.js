import React from 'react';
import { render, cleanup } from '@testing-library/react';
import 'jest-dom/extend-expect';
import BetList from '../components/BetList';

afterEach(cleanup);

it('renders', () => {
    const { asFragment } = render(<BetList bets={[
        {
            cityId: '123',
            name: 'test',
            inzet: 12,
            guess: 22,
            time: 'date',
            timeOfNow: 'date2'
        }
    ]} />);
    expect(asFragment()).toMatchSnapshot();
});