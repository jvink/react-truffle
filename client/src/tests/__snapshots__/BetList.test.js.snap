// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="card col-4"
  >
    <h2>
      Wallet
    </h2>
    <p>
      The stored value is: 
    </p>
    <h4>
      Accounts: 
    </h4>
    <ul>
      <li />
    </ul>
    <button
      class="btn btn-secondary mb-4"
      type="button"
    >
      click here
    </button>
    <h4>
      Your bets:
    </h4>
  </div>
</DocumentFragment>
`;
